src/super_gradients/training/models/segmentation_models/common.py

```diff
@@ -1,7 +1,20 @@
 import torch.nn as nn
+from abc import ABC, abstractmethod
+
+from typing import List
+
 from super_gradients.modules import ConvBNReLU
 
 
+class AbstractSegmentationBackbone(nn.Module, ABC):
+    @abstractmethod
+    def get_backbone_output_number_of_channels(self) -> List[int]:
```

Comment:
Do we want to also have a method `get_backbone_output_strides(self) -> List[int]`

Or maybe even better to make it a bit future-proof:

```
@dataclass
class FeatureMapOutputSpec:
   channels: int
   stride: int
   
def get_output_spec() -> Tuple[FeatureMapOutputSpec, ...]
```

Reply:
As I said, there's a lot to improve there :)
I think we can do this one later as part of an additional refactoring

Reply:
I agree with @BloodAxe , for this backbone to be a general case for segmentation models, the out strides must be exposed. So that decoder as in STDC or PPLiteSeg could assert for a valid backbone or modify the decoder accordingly

Reply:
Done, [519d66c](https://github.com/Deci-AI/super-gradients/pull/704/commits/519d66c527525dedd8546fec3bd26716903d1c87)